### ISSUE TITLE
[feature] more shell completions

### ISSUE THREAD

**Issue by Freed-Wu** [jgm/pandoc#8542]
**Describe your proposed improvement and the problem it solves.**

Now `pandoc --bash-completion` can generate bash completion script. Why not provide more `--XXsh-completion`?
For example, zsh has a [completion script](https://github.com/zsh-users/zsh/blob/master/Completion/Unix/Command/_pandoc)
for pandoc, however, it is written manually:

```zsh
_arguments -s \
  {-f+,-r+,--from=,--read=}'[specify input format]: :_pandoc_format -T input' \
  {-t+,-w+,--to=,--write=}'[specify output format]: :_pandoc_format -T output' \
  {-o+,--output=}'[write output to FILE instead of stdout]:file:_files' \
  '--data-dir=[specify the user data directory to search for pandoc data files]:data directory:_files -/' \
  {-d+,--defaults=}'[read default from YAML file]: :_pandoc_defaults_file' \
  '--shift-heading-level-by=[shift heading levels by specified number]:positive or negative integer: ' \
  '!--base-header-level=:number [1]:(1 2 3 4 5)' \
  '!--strip-empty-paragraphs[deprecated. Use the +empty_paragraphs extension instead]' \
  '--indented-code-classes=[classes to use for indented code blocks]:class list (comma-separated)' \
  '--default-image-extension=[specify a default extension to use when image paths/URLs have no extension]:extension: ' \
  '--file-scope[parse each file individually before combining for multifile documents]' \
  ...
```

When the option of pandoc changed, the maintainers of zsh have to update it manually. If pandoc can provide `--zsh-completion` which use this file as a template but update automatically, it will be greater.

TIA!

Same issues:
- https://github.com/sharkdp/vivid/issues/86
- https://github.com/sharkdp/hexyl/issues/155
- https://github.com/rhysd/actionlint/issues/300
- https://github.com/mvdan/sh/issues/914

**Comment by jgm** [comment]
As noted in the linked issue: I don't want to integrate a completion script that is mostly manually written. If it can be done in a template form like the bash completions, we could consider it.

**Comment by Freed-Wu** [comment]
> If it can be done in a template form like the bash completions, we could consider it.

zsh support completing the descriptions of options, that is:

For zsh:

```
❯ pandoc --<TAB>
option
--abbreviations              specifies a custom abbreviations file
--ascii                      use only ASCII characters in output, supported only for HTML and DocBook output
--bash-completion            generate a bash completion script
--biblatex                   use biblatex for citations in LaTeX output
...
```

For bash which don't support descriptions of options.

```
$ pandoc --<TAB>
--ascii
--atx-headers
--base-header-level
--bash-completion
--biblatex
...
```

However, the descriptions of options exists in the
[man page](https://github.com/jgm/pandoc/blob/main/man/pandoc.1), which is generated
from a markdown file [MANUAL.txt](https://github.com/jgm/pandoc/blob/main/MANUAL.txt) by pandoc.

However `pandoc --zsh-completion` cannot access `MANUAL.txt`. I think use
a script to handle `MANUAL.txt` to generate zsh completion script `_pandoc` is
a little weird. Otherwise users have to give up the descriptions of options.

**Comment by jgm** [comment]
Moreover, zsh wants concise half-line descriptions, and MANUAL.txt won't provide those.

**Comment by Freed-Wu** [comment]
> zsh wants concise half-line descriptions, and MANUAL.txt won't provide those.

Right. It looks like before this problem is fixed, the support for zsh compleiton is hard to realize.

**Comment by jidanni** [comment]
With pandoc 3.1.11.1 on Debian, still:
```
$ cat --<TAB>
--help  --show-tabs  --version ...
$ pandoc --<TAB>
```
Nothing.

**Comment by jgm** [comment]
@jidanni Isn't that a Debian packaging issue rather than anything specific to this project?
We do provide a way to get shell completions, but the system has to be set up to integrate them....

**Comment by jidanni** [comment]
OK, I filed <DEL>https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1121654</del>

**Comment by jidanni** [comment]
By the way, I am now installing the debs directly from
https://github.com/jgm/pandoc/releases/tag/3.9.0.2
so there is no maintainer middleman.

We read "Use debhelper and put package.bash-completion in your debian/ directory and include dh-sequence-bash-completion in your Build-Depends...

Here is another package from **outside** Debian that works fine:
```
$ dlocate complet|grep pan
panache: /usr/share/bash-completion
panache: /usr/share/bash-completion/completions
panache: /usr/share/bash-completion/completions/panache
panache: /usr/share/zsh/vendor-completions...
$ panache <TAB>
-q                --config          --quiet           --jobs            lsp
-v                --stdin-filename  --verbose         --help            lint
-j                --flavor          --isolated        --version         clean
-h                --color           --no-cache        format            debug
-V                --no-color        --cache-dir       parse             help
```
Based on that I will close the Debian bug report, as it is clearly the task of the .deb creator.

**Comment by jgm** [comment]
Don't do that.  Debian doesn't just use my deb package; they have a completely separate packaging process, and their package has nothing to do with mine. They are free to add bash completion if they like.

**Comment by jidanni** [comment]
I am not using Debian's .deb these days. I am now using yours directly. Just as I am using panache directly. The Panache .Deb has the completions working fine.
